src/language/classes.md

```diff
@@ -195,6 +192,29 @@ void main() {
 }
 ```
 
+Initializing a non-`late` instance variable where it's declared
+sets the value when the instance is created,
+before the constructor and its initializer list execute.
+As a result, the initializing expression (after the `=`)
+of a non-`late` instance variable can't access `this`.
+
+<?code-excerpt "misc/lib/language_tour/classes/point_this.dart (this-late)"?>
+```dart
+double initialX = 1.5;
+
+class Point {
+  double? x = initialX;
+  // OK, can access declarations that do not depend on `this`.
+  double? y = this.x;
+  // ERROR, can't access `this` in non-`late` initializer.
+  late double? z = this.x;
+  // OK, can access `this` in `late` initializer.
+
+  Point(this.x, this.y);
+  // OK, not an initializing expression, can access `this`.
+}
+```
```

Comment:
Similar comments as in examples/misc/lib/language_tour/classes/point_this.dart. (I guess that library and this snippet are kept in sync automatically).

Reply:
Yeah they're linked, and since it's happening on Save I can't manually overwrite.